### ISSUE TITLE
Add overloaded constructors

### ISSUE THREAD

**Issue by AlexanderJDupree** [AlexanderJDupree/LinkedListsCPP#6]
- [x] fill constructor

Ex. 

```c++
LinkedList<std::string> list (4, "hi");
```
Constructs a list with 4 strings with a value of "hi"

- [x] range constructor

Ex. 

```c++
std::vector<int> nums(4, 100);
LinkedList<int> list(nums.begin(), nums.end());
```
iterates through nums and constructs a list from a copy of each element in nums

- [x] initializer list constructor

Ex. 

```c++
LinkedList<std::string> list { "Hello", "world"};
```
constructs a list with the values "Hello" & "world" as the elements.

- [x] copy constructor

Ex.

```c++
LinkedList<int> first { 1, 2, 3, 4 };
LinkedList<int> second(first);
```

Creates a copy of first into second

**Comment by jake-bickle** [comment]
I'll work on making the copy constructor, which I'll also implement with the overloaded assignment operator.

**Comment by AlexanderJDupree** [comment]
Copy constructor implemented in PR #28

**Comment by AlexanderJDupree** [comment]
Initializer list constructor implemented in PR #37
